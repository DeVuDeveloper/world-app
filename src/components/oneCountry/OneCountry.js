import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { BsArrowRightCircle } from 'react-icons/bs';
import CountryCode from './CountryCode';

import './oneCountry.css';

const OneCountry = ({
  id,
  countryName,
  todayConfirmed,
}) => {
  const countryImg = CountryCode[countryName];
  const map = countryImg
    ? `https://raw.githubusercontent.com/VuDej/worldMaps/main/maps/${countryImg.toLowerCase()}/128.png`
    : '';

  return (

    <li
      key={id}
      className="card"
    >

      <Link
        to={`/${id}`}
        style={{ textDecoration: 'none' }}
      >
        <div>

          <span className="card-images">
            <img
              className="card-img"
              src={map}
              alt="country"
              width="80"
              height="80"
            />
            <BsArrowRightCircle className="circle" />
          </span>
          <span className="card-text">
            <h3>{countryName}</h3>
            <p>{todayConfirmed}</p>
          </span>

        </div>

      </Link>

    </li>

  );
};
OneCountry.propTypes = {
  id: PropTypes.string.isRequired,
  countryName: PropTypes.string.isRequired,
  todayConfirmed: PropTypes.number.isRequired,
};
export default OneCountry;
