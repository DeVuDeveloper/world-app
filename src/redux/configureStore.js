import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import globalReducer from './duckFolder/duckFile';

const reducer = combineReducers({
  globalReducer,
});
const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;
