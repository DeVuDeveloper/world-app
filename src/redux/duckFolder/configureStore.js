import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { globalReducer } from './reducers/homepage';

const reducer = combineReducers({
  globalReducer,
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;