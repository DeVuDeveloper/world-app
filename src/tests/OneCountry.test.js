import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import OneCountry from '../components/oneCountry/OneCountry';

const data = [
  {
    id: 'afghanistan',
    name: 'Afghanistan',
    today_confirmed: 4,
  },
  {
    id: 'angola',
    name: 'Angola',
    today_confirmed: 4,
  },
];

const initialState = [];
const mockStore = configureStore();
let store;

it('renders correctly', () => {
  store = mockStore(initialState);

  const oneCountry = data.map((country) => (
    <Provider store={store} key={country.id}>
      <BrowserRouter>
        <OneCountry />
      </BrowserRouter>
    </Provider>
  ));

  const tree = renderer.create(oneCountry).toJSON();
  expect(tree).toMatchSnapshot();
});
