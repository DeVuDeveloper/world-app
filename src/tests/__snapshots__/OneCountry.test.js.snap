// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <li
    className="card"
  >
    <a
      href="/undefined"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <span
        className="card-images"
      >
        <img
          alt="country"
          className="card-img"
          height="80"
          src=""
          width="80"
        />
        <svg
          className="circle"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="card-text"
      >
        <h3 />
        <p />
      </span>
    </a>
  </li>,
  <li
    className="card"
  >
    <a
      href="/undefined"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <span
        className="card-images"
      >
        <img
          alt="country"
          className="card-img"
          height="80"
          src=""
          width="80"
        />
        <svg
          className="circle"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <span
        className="card-text"
      >
        <h3 />
        <p />
      </span>
    </a>
  </li>,
]
`;
